v1/plugin/session.go

```diff
@@ -32,17 +30,28 @@ type Arg struct {
 
 	// enable pprof
 	Pprof bool
+
+	// Path to TLS certificate file for a TLS server
+	CertPath string
+
+	// Path to TLS private key file for a TLS server
+	KeyPath string
+
+	// Flag requesting server to establish TLS channel
+	TLSEnabled bool
 }
 
 // getArgs returns plugin args or default ones
-func getArgs() error {
+func getArgs() (*Arg, error) {
 	pluginArg := &Arg{}
-	if os.Args[1] == "" {
-		return nil
+	osArgs := libInputOutput.readOSArgs()
+	var paramStr = osArgs[1]
+	if paramStr == "" {
+		paramStr = "{}"
 	}
-	err := json.Unmarshal([]byte(os.Args[1]), pluginArg)
+	err := json.Unmarshal([]byte(paramStr), pluginArg)
 	if err != nil {
-		return err
+		return nil, err
```

Comment:
Consider adding of  `pluginArg` to error message, it should be easier to debug user's issues.